src/main/java/seedu/address/model/food/Macronutrient.java

```diff
@@ -14,29 +15,56 @@
      * @param amount
      * @param caloricMultiplier
      */
-    public Macronutrient(String name, int amount, int caloricMultiplier) {
-        requireAllNonNull(name, amount, caloricMultiplier);
-        assert amount >= 0 : "Negative Macronutrient Amount";
+    public Macronutrient(String type, int amount, int caloricMultiplier) {
+        requireAllNonNull(type, amount, caloricMultiplier);
+
+        assert !type.equals("") : "Name cannot be blank";
```

Comment:
Assertion here could be more descriptive (e.g. "Macronutrient type cannot be blank"), since we are changing `name` to `type`.

Reply:
Fixed